### DISCUSSION TITLE
Immutable-Infrastructure-as-Code (IIaC) workspace for FPrime (project) development

### DISCUSSION THREAD

**Discussion by capsulecorplab** [nasa/fprime#2141] [Show and tell]
Hello FPrime community,
I've been working on an Immutable-Infrastructure-as-Code (IIaC) development environment / workspace image for developing FPrime projects. Just published an initial release (v0.1.0) and wanted to share with the FPrime community. Current workspace image contains development tools for building FPrime projects with FPrime v3.2.0. Intent for upcoming release(s) will be to add support for community projects (s.a., the led blinker, hello world, system reference, and baremetal reference). Hope this helps accelerate some of the development time with FPrime projects. If it meets the community standards, I'd also be happy to eventually migrate the repo to the fprime-community GitHub organization. Let me know what you think~

EDIT: repo has been migrated to https://github.com/fprime-community/fprime-workspace-image

![fprime-workspace-image screenshot](https://raw.githubusercontent.com/capsulecorplab/fprime-workspace-image/main/screenshot-of-fprime-workspace-image-with-ingenuity-above-perseverance-on-mars-wallpaper.png)
[Ingenuity Helicopter above Perseverance on Mars](https://photojournal.jpl.nasa.gov/catalog/PIA24127) (wallpaper) image credit: NASA/JPL-Caltech

**Comment by Joshua-Anderson** [comment]
Very cool! This could be a good option for F' workshops. Often with F' workshops a large portion of the class is helping everyone get their environments working and debugging machine specific/config issues.

A consistent environment people could just launch and start working in could help streamline the workshop

**Reply by capsulecorplab** [reply]
Absolutely! That's the whole idea with this approach; you virtually have the same working environment that runs cross-platform with minimal setup/configuration :slightly_smiling_face:

**Comment by capsulecorplab** [comment]
Just released [v0.2.0](https://github.com/capsulecorplab/fprime-workspace-image/releases/tag/v0.2.0) of the workspace which includes Arduino build tools for `fprime-arduino` projects :slightly_smiling_face:

**Comment by capsulecorplab** [comment]
Update: I've recently published a pre-release of the `fprime-workspace-image` to support building the [baremetal reference](https://github.com/fprime-community/fprime-baremetal-reference) with fprime v3.4.0 (with much assistance and thanks to @ethancheez) [here](https://github.com/capsulecorplab/fprime-workspace-image/releases/tag/v3.4.0-alpha1.1.0)! I'm preparing to [migrate the repo to the fprime-community GitHub organization](https://github.com/capsulecorplab/fprime-workspace-image/issues/33) and am wondering if there is a process for doing so; thoughts @thomas-bc @LeStarch ?

**Reply by thomas-bc** [reply]
Responded in https://github.com/capsulecorplab/fprime-workspace-image/issues/33 !

**Reply by thomas-bc** [reply]
Repository has been transferred to F´ Community

**Reply by capsulecorplab** [reply]
Thanks again @thomas-bc ! When convenient, it seems we also need to enable GitHub Container Registry (GHCR). Otherwise, the CI is broken. I've logged the issue [here](https://github.com/fprime-community/fprime-workspace-image/issues/47)
